compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs

```diff
@@ -617,6 +620,33 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                     });
 
                 if let Some((decl, body_id)) = closure_decl_and_body_id {
+                    let mut visitor = UnresolvedTypeFinder::new(self);
+                    for arg in fn_sig.inputs().skip_binder().iter() {
+                        if let ControlFlow::Break((ty, _)) = visitor.visit_ty(*arg) {
+                            let mut inner = self.inner.borrow_mut();
+                            let ty_vars = &inner.type_variables();
```

Comment:
Can you inline this variable?

Reply:
I cannot due to the lifetime of the `borrow_mut`:
```
error[E0716]: temporary value dropped while borrowed
   --> compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs:626:42
    |
626 | ...                   let ty_vars= self.inner.borrow_mut().type_variables();
    |                                    ^^^^^^^^^^^^^^^^^^^^^^^                 - temporary value is freed at the end of this statement
    |                                    |
    |                                    creates a temporary which is freed while still in use
...
629 | ...                           let var_origin = ty_vars.var_origin(ty_vid);
    |                                                -------------------------- borrow later used here
    |
    = note: consider using a `let` binding to create a longer lived value
```

Reply:
sorry, I mean inlining `ty_vars` -- like `let var_origin = self.inner.borrow_mut().type_variables().var_origin(ty_vid)`?